controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -52,11 +84,439 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+			}
+		}
+	}
+
+	r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateConfiguring
+
+	if err := r.reconcileCephBlockPool(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephFilesystemSubVolumeGroup(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephClientRBDProvisioner(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephClientRBDNode(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephClientCephFSProvisioner(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephClientCephFSNode(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephClientHealthChecker(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !r.areRookCephResourcesReady() {
+		return reconcile.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
+	}
+
+	r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateReady
+	r.storageConsumer.Status.GrantedCapacity = r.storageConsumer.Spec.Capacity
 
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) reconcileCephBlockPool() error {
+
+	storageClusterList := ocsv1.StorageClusterList{}
+	err := r.Client.List(r.ctx, &storageClusterList, client.InNamespace(r.namespace))
+	if err != nil {
+		return fmt.Errorf("Error fetching StorageClusterList. %+v", err)
+	}
+
+	var failureDomain string
+	if len(storageClusterList.Items) == 1 {
+		failureDomain = storageClusterList.Items[0].Status.FailureDomain
+	}
+
+	capacity := r.storageConsumer.Spec.Capacity.String()
+	desired := &rookCephv1.CephBlockPool{
+		Spec: rookCephv1.NamedBlockPoolSpec{
+			PoolSpec: rookCephv1.PoolSpec{
+				FailureDomain: failureDomain,
+				Replicated: rookCephv1.ReplicatedSpec{
+					Size:                     3,
+					ReplicasPerFailureDomain: 1,
+					TargetSizeRatio:          .49,
+				},
+				EnableRBDStats: false,
+				Quotas: rookCephv1.QuotaSpec{
+					MaxSize: &capacity,
+				},
+			},
+		},
+	}
+
+	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephBlockPool, func() error {
+		if err := r.own(r.cephBlockPool); err != nil {
+			return err
+		}
+
+		r.cephBlockPool.Spec = desired.Spec
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephBlockPool.", "CephBlockPool", klog.KRef(r.cephBlockPool.Namespace, r.cephBlockPool.Name))
+		return err
+	}
+
+	r.storageConsumer.Status.CephObjects.BlockPoolName = r.cephBlockPool.Name
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephFilesystemSubVolumeGroup() error {
+	cephFilesystemList := rookCephv1.CephFilesystemList{}
+	err := r.Client.List(r.ctx, &cephFilesystemList, client.InNamespace(r.namespace))
+
+	if err != nil {
+		return fmt.Errorf("Error fetching CephFilesystemList. %+v", err)
+	}
+
+	var cephFileSystemName string
+	if len(cephFilesystemList.Items) == 1 {
+		cephFileSystemName = cephFilesystemList.Items[0].Name
+	}
```

Comment:
Logic change request:
Less than 1 -> error
More than 1 -> warning
Take the first